.github/workflows/release-python.yml

```diff
@@ -155,13 +166,20 @@ jobs:
         name: wheels
         path: dist
 
-  release:
-    name: Release
+  publish:
     runs-on: ubuntu-latest
+    # `needs` forces this job to wait until all specified jobs
+    # are finished to run. Typically, those jobs would all have
+    # to be successful, but when combined with `if: always()`,
+    # this job is allowed to run after all the needed jobs
+    # finish, regardless of their outcome. In this case, we
+    # still make sure that at least a source distribution
+    # can be published.
     needs: [ macos, linux, windows, sdist ]
+    if: always() && needs.sdist.result == 'success'
```

Comment:
Is the `always()` necessary if you have the `needs.sdist.result` bit? I can see it having a side effect that is needed, but from a boolean expression standpoint it seems pointless to have what is effectively `true &&`

Reply:
Yeah it's weird, but [`always()`](https://docs.github.com/en/actions/learn-github-actions/expressions#always) is special, it doesn't just mean `true` here. It means always run the job. On its own, `needs` requires all specified jobs to finish successfully. That would be a problem, because we're making some jobs optional and "skipped" doesn't count as a success. We fix this by combining `needs` with `if: always()`, which makes it so the jobs specified in needs just need to be finished before the job can run, they don't need to be successful. We can add `needs.sdist.result == 'success'` to ensure there is always at least an sdist available to publish, but it would effectively do nothing on it's own.